library/deletedrug.php

```diff
@@ -27,11 +27,46 @@
     if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
         CsrfUtils::csrfNotVerified();
     }
+
+    /**
+     * find the drug name in the prescription table
+     */
+    try {
+        $drug_name = "SELECT patient_id, drug FROM prescriptions WHERE id = ?";
+        $dn = sqlQuery($drug_name, [$id]);
+    } catch (Exception $e) {
+        echo 'Caught exception ', text($e->getMessage()), "\n";
+        if ($e->getMessage()) {
+            exit;
+        }
+    }
+
+    /**
+     * remove drug from the medication list if exist
+     */
+    try {
+        $pid = $dn['patient_id'];
+        $drugname = $dn['drug'];
+        $medicationlist = "DELETE FROM lists WHERE pid = ? AND title = ?";
```

Comment:
note this would also remove an allergy of same name.
(`type` needs to be medication )

Reply:
I thought about adding medication as a field. But in my limited knowledge, the drug name is exact so unless the user enters  Lisinopril 10 MG Oral Tablet - Oral Tablet - Oral Pill - 10 mg as an allergy. I can see where the allergy would be removed also. But if they are allergic to lisinopril and that is entered as an allergy and the prescription drug only has lisinopril then yes. It will delete both.